pkg/inventory/inventory-client-factory.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package inventory
+
+import cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+var (
+	_ InventoryClientFactory = ClusterInventoryClientFactory{}
+)
+
+type InventoryClientFactory interface {
+	NewInventoryClient(factory cmdutil.Factory) (InventoryClient, error)
+}
+
+type ClusterInventoryClientFactory struct {
```

Comment:
Let's add some comments to describe this concrete factory. I think we should change the name from `Cluster` to `ConfigMap`, since the default inventory object is a `ConfigMap`. But we can debate that for later.